core/src/scene/scene.h

```diff
@@ -32,6 +33,10 @@ class Scene {
     Scene();
     ~Scene();
 
+    using ArrayValue = std::vector<double>;
+    using Value = variant<none_type, bool, double, std::string, ArrayValue>;
```

Comment:
This is not sufficient to encode the possible values of a global property entry. Elements of `global` can be arbitrary yaml, including sequences of strings and sequences of booleans. It seems much simpler and more natural to store the actual yaml node for each entry (or a pointer to it). 

Tangentially, I would recommend that we not introduce any more types called `Value` - even one type with a meaningless name is plenty for any given project ;)